modules/references/client/components/ListReferences.component.js

```diff
@@ -14,28 +14,58 @@ import ReferencesSection from './read-references/ReferencesSection';
  */
 export default function ListReferences({ profile, authenticatedUser }) {
   const { t } = useTranslation('references');
-  const [publicReferences, setPublicReferences] = useState([]);
-  const [pendingReferences, setPendingReferences] = useState([]);
+  const [publicExperiencePairs, setPublicExperiencePairs] = useState([]);
+  const [pendingExperiences, setPendingExperiences] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
+  function pairUpExperiences(experiences) {
+    /* TODO: compare by ids (couldn't make it work, so I used usernames to compare for now) */
```

Comment:
Is there any benefit in doing this at client side? We could do it at the backend already and get API response in a format the client needs it.

Fine for this PR but perhaps something we can revisit in separate PR. :-)